### PR TITLE
tests(customflags): Improve testing suite

### PR DESCRIPTION
**Pull request by JeremiGendron**
Part of #1325 

---

# Description

I have found that the testing suite for the cmd package's customflags was incomplete.

I am working on adding tests for everything provided by the customflags.go file.

 - Test against fixed array of DirectoryStrings
 - Provide globals for redundant variables
 - Refactor current single test to use new setup
 - Add tests for DirectoryString.Set(string) method
 - Add tests for homeDir() top-level func

### CODE REVIEW

File: shared/cmd/customflags_test.go [hunk 1]
```diff
@@ -22,20 +22,67 @@ import (
...
 )
 
+var (
+  // user-specific tests can use the usr variable
+  usr, _ = user.Current()
+  against []DirectoryString
+)
+
+func init() {
+  // test for environment variables in paths with $DDDXXX
```
Comment (rauljordan) [inline shared/cmd/customflags_test.go]: Use punctuation at end of Go comments.

File: shared/cmd/customflags_test.go [hunk 2]
```diff
@@ -22,20 +22,67 @@ import (
...
+    original := test.Value
+    test.Set("$DDDXXX")
+    got := test.Value
+    if test.Value != expected {
+      t.Errorf("test set path %s, got %s, expected %s\n", original, got, expected)
+    }
+  }
+}
+
+func TestStringDirectoryString(t *testing.T) {
```
Comment (nisdas) [inline shared/cmd/customflags_test.go]: Why do you have an empty function ?

Comment (JeremiGendron) [reply shared/cmd/customflags_test.go]: Work in progress, I will remove it for now and continue it later

**Comment by JeremiGendron** [comment]
Are there any guidelines I should be sure to understand before resuming work here?

**Comment by nisdas** [comment]
@JeremiGendron Here are our contribution guidelines https://github.com/prysmaticlabs/prysm/blob/master/CONTRIBUTING.md. We are planning to have a more detailed coding guide in the future.

File: shared/cmd/customflags_test.go [hunk 3]
```diff
@@ -22,20 +22,89 @@ import (
...
+func init() {
+  // test for environment variables in paths with $DDDXXX.
+  os.Setenv("DDDXXX", "/tmp")
+  // test against valid DirectoryString structs.
+  against = []DirectoryString{
+	  {"/home/someuser/tmp"},
+	  {usr.HomeDir + "/tmp"},
+	  {"~thisOtherUser/b"},
+    {"/tmp/a/b"},
+    {"/a/b"},
```
Comment (prestonvanloon) [inline shared/cmd/customflags_test.go]: Formatting; please run gofmt

File: shared/cmd/customflags_test.go [hunk 4]
```diff
@@ -22,20 +22,89 @@ import (
...
+
+func TestSetDirectoryString(t *testing.T) {
+  tests := make([]DirectoryString, len(against))
+  copy(tests, against)
+  expected := "/tmp"
+  for _, test := range tests {
+    original := test.Value
+    test.Set("$DDDXXX")
+    got := test.Value
+    if test.Value != expected {
```
Comment (prestonvanloon) [inline shared/cmd/customflags_test.go]: Use got here

File: shared/cmd/customflags_test.go [hunk 5]
```diff
@@ -22,20 +22,89 @@ import (
...
+}
+
+func TestPrefixedNames(t *testing.T) {
+  test := "d,datadir,,n,network"
+  expected := "-d, --datadir, --, -n, --network"
+
+  prefixed := prefixedNames(test)
+
+  if prefixed != expected {
+    t.Errorf("test prefixed names for %s, got %s, expected %s\n", test, prefixed, expected)
```
Comment (prestonvanloon) [inline shared/cmd/customflags_test.go]: You don’t need new line at end of message for tests

File: shared/cmd/customflags_test.go (hunk 3 above)
Comment (JeremiGendron) [reply shared/cmd/customflags_test.go]: thanks

**Comment by JeremiGendron** [comment]
not in-progress anymore!

File: shared/cmd/customflags_test.go [hunk 6]
```diff
@@ -22,20 +22,89 @@ import (
 	"testing"
 )
 
+var (
+	// user-specific tests can use the usr variable.
+	usr, _  = user.Current()
+	against []DirectoryString
+)
+
+func init() {
```
[the remark this answers is not in the document]
Comment (nisdas) [reply shared/cmd/customflags_test.go]: ah, k thanks didnt know we could do that

**Comment by JeremiGendron** [comment]
messed it up!

**Comment by nisdas** [comment]
@JeremiGendron Why did you close this PR ? You just had to address a comment, then we could have merged it in

**Pull request closed without merging**